src/igvfd/audit/sample.py

```diff
@@ -35,3 +35,33 @@ def audit_sample_sorted_fraction_parent_child_check(value, system):
         )
         if prop_errors != '':
             yield AuditFailure('inconsistent sorted fraction metadata', detail, level='ERROR')
+
+
+@audit_checker('Sample', frame='object')
+def audit_sample_virtual_donor_check(value, system):
+    '''Non-virtual samples should not be linked to virtual donors.'''
+    if 'virtual' and 'donors' in value:
+        sample_id = value['@id']
+        sample_virtual = value['virtual']
+        donor_ids = value.get('donors')
+        donors_error = []
+        for d in donor_ids:
+            donor_object = system.get('request').embed(d + '@@object')
```

Comment:
i think you can also skip calculated.

Reply:
i had that initially, but i couldn't produce `detail` object; skipping calculated i couldn't get value['@id'].